### ISSUE TITLE
GLPI10.0.2 do not receive the ticket.(Oauth IMAP application) Pages:1

### ISSUE THREAD

**Issue by jjomi** [glpi-project/glpi#12446]
### Code of Conduct

- [ ] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [ ] I have searched the existing issues

### Version

10.0.2

### Bug description

Hi,

I connect to my gmail bay Oauth IMAP applicaion Plugin.
I suceess to create an authorization and created receiver.
but i can't receive the ticket and error occur.

![1 oauth](https://user-images.githubusercontent.com/89990385/184832030-60a28767-ea23-4244-9fe2-311fe122d75f.jpg)
![oauth2](https://user-images.githubusercontent.com/89990385/184832126-7d8ecf9a-4a39-4034-9af3-54c2f483f8d5.png)

>> Oauth IMAP status
Oauth Provider : Google
-> authentication is seccuss

![receiver](https://user-images.githubusercontent.com/89990385/184832148-1608cd89-bc78-4e51-b43f-7b2a5a305d25.png)

>> my receiver
ID@gmail.com
Connection option : Oauth IMAP / SSL
Incomin mail folder : INBOX
Login : ID@gmail.com

>> try 'get mail ticket now' : false
error mwssage :
An error occurred trying to connect to the collector.
cannot connect to host; error = stream_socket_client(): unable to connect to ssl://mail.google.com:993 (Connection timed out) (errno = 0 )

![php-error](https://user-images.githubusercontent.com/89990385/184832203-d84af2a9-c943-4ddd-ab88-0cf85ecb3ab1.png)

>>php error log
[2022-08-12 16:59:52] glpiphplog.CRITICAL:   *** Uncaught Exception Laminas\Mail\Protocol\Exception\RuntimeException: cannot connect to host; error = stream_socket_client(): unable to connect to ssl://mail.google.com:993 (Connection timed out) (errno = 0 ) in /var/www/glpi/vendor/laminas/laminas-mail/src/Protocol/ProtocolTrait.php at line 100
  Backtrace :
  ...thimap/inc/imap/imapoauthprotocol.class.php:120 Laminas\Mail\Protocol\Imap->setupSocket()
  ...authimap/inc/imap/imapoauthstorage.class.php:74 GlpiPlugin\Oauthimap\Imap\ImapOauthProtocol->connect()
  ...authimap/inc/mailcollectorfeature.class.php:101 GlpiPlugin\Oauthimap\Imap\ImapOauthStorage->__construct()
  src/Toolbox.php:2104                               GlpiPlugin\Oauthimap\MailCollectorFeature::GlpiPlugin\Oauthimap\{closure}()
  src/MailCollector.php:1369                         Toolbox::getMailServerStorageInstance()
  src/MailCollector.php:692                          MailCollector->connect()
  front/mailcollector.form.php:103                   MailCollector->collect()

I think gmail authorization was connected.
why this error occured?

Glpi version : 10.0.2
os verion : ubuntu 20.0 (aws ec2)
php 7.4
mariaDB : 10.3.34

in security group, imap, imaps and smtp port is opend.

**Comment by tomolimo** [comment]
Hello @jjomi 
Do you still have this issue?
Could you try for the server FQDN `imap.gmail.com` instead of `mail.google.com`?
I made a test with these settings and with oauth2 of course, and it worked like that.
Thank you,
Regards,
Tomolimo

**Comment by pierlaze** [comment]
I have the same problem, connection error 1, could someone solve it? I have Oauth configured with Azure. Out of nowhere it stopped working, your help please
GLPI 9.5.7
![image](https://user-images.githubusercontent.com/106604012/232805124-622dc6cc-7ec6-440b-b97c-749a4f893824.png)
![image](https://user-images.githubusercontent.com/106604012/232805539-ec6b1cb0-f041-4925-8a4d-3960d0cf8db8.png)

**Comment by ldgmaia** [comment]
Hi @pierlaze
I am having the same issue. Out of nowhere, I am getting this error using the oauth. I just updated glpi and the oauthimap plugin to the latest version (10.0.9 and 1.4.3 respectively). Have you fix this problem? I am totally lost since I am using docker and I am having a hard time trying to find any error log.

**Comment by ldgmaia** [comment]
I just figured it out. My oauth/imap settings was set to use TLS, and have being working like this forever. Yesterday stopped working. I checked that the Diagnose button on the oauthimap plugin was workiong, and not using TLS. I just removed the TLS setting and started working again.
![image](https://github.com/glpi-project/glpi/assets/3820420/f321b430-2bca-4452-a195-3d1eb8b34a65)

I hope this info can help someone!
Thank you.
